core/src/main/resources/static/js/browseAcls.js

```diff
@@ -415,95 +415,6 @@ app.controller("browseAclsCtrl", function($scope, $http, $location, $window) {
-
-                var serviceInput = {};
-                $scope.alertTopicDelete = null;
-                $scope.alert = null;
-
-                if(!$scope.partitionsSelected || $scope.partitionsSelected == 'selected'){
-                    $scope.alertnote = "Please select topic partitions.";
-                    $scope.alert = $scope.alertnote;
-                    $scope.showAlertToast();
-                    return;
-                }
-
-                var tmpPartitionsSelected = $scope.partitionsSelected;
-                var tmpRepsSelected = $scope.replicationFactorSelected;
-
-                // selecting default partitions
-                if(tmpPartitionsSelected.indexOf("default") > 0)
-                {
-                    tmpPartitionsSelected = tmpPartitionsSelected.replace(" (default)","");
-                }
-
-                // selecting default rf
-                if(tmpRepsSelected.indexOf("default") > 0)
-                {
-                    tmpRepsSelected = tmpRepsSelected.replace(" (default)","");
-                }
-
-                serviceInput['environment'] = envSelected;
-                serviceInput['topicname'] = $scope.topicSelectedParam;
-                serviceInput['topicpartitions'] = tmpPartitionsSelected;
-                serviceInput['replicationfactor'] = tmpRepsSelected;
-                serviceInput['teamname'] = $scope.teamname;
-                serviceInput['appname'] = "App";//$scope.addTopic.app;
-                serviceInput['remarks'] = "Topic promotion."
-                serviceInput['topictype'] = 'Create';
-                serviceInput['description'] = $scope.topicSelectedParam + " topic."
-
-                $http({
-                    method: "POST",
-                    url: "createTopics",
-                    headers : { 'Content-Type' : 'application/json' },
-                    params: {'addTopicRequest' : serviceInput },
-                    data: serviceInput
-                }).success(function(output) {
-                    if(output.result == 'success'){
-                        swal({
-                        	 title: "",
-                        	 text: "Topic Promotion Request : "+output.result,
-                        	 showConfirmButton: true
-                         }).then(function(isConfirm){
-                                $window.location.href = $window.location.origin + $scope.dashboardDetails.contextPath + "/myTopicRequests?reqsType=created&topicPromotionCreated=true";
-                          });
-                    }
-                    else{
-                            $scope.alert = "Topic Promotion Request : "+output.result;
-                            $scope.showSubmitFailed('','');
-                        }
-                }).error(
-                    function(error)
-                    {
-                        $scope.handleValidationErrors(error);
-                    }
-                );
-
-            };
-
```

Comment:
I don't see a replacement for this method, to set firstPromote to true.
Without this I think in browseAcls.html the option will not ever be presented to the user to promote the topic?


or is this section of code in browseAcls.html now completly obsolete as we pass the user to the create topic page?


```
<td align="left" width="25%">
											<div ng-show="firstPromote == 'true'" class="form-group has-success">
												<select class="form-control custom-select"
														required
														ng-model="partitionsSelected"
														ng-options="partitions as partitions for partitions in promotionParams.partitionsList">
													<option value="" selected="selected">
														Select Partitions</option>
												</select>
											</div>
										</td>
										<td align="left" width="25%">
											<div ng-show="firstPromote == 'true'" class="form-group has-success">
												<select class="form-control custom-select"
														required
														ng-model="replicationFactorSelected"
														ng-options="replicas as replicas for replicas in promotionParams.replicationFactorList">
													<option value="" selected="selected">
														Select Replication Factor</option>
												</select>
											</div>
										</td>
```

Reply:
As we have moved the promote logic , redirecting user to topic req page, these methods are not required anymore. 
I should have deleted them earlier.